### PR TITLE
[tests-only][full-ci]Add API step to upload file in personal space

### PR DESCRIPTION
**Pull request by SagarGi**
### Description
This PR adds API steps to upload files in All files (`oc10`) and in personal space for (`oCIS`).

### Related issue
https://github.com/owncloud/web/issues/8101

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oC10</strong> https://drone.owncloud.com/owncloud/web/31644/11/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/31644/tracing/public-link-alice-2023-1-16-09-27-11.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/31644/tracing/public-link-anonymous-2023-1-16-09-27-27.zip 


</p></details>

File: tests/e2e/support/api/davSpaces/spaces.ts [hunk 1]
```diff
@@ -51,3 +51,42 @@ export const createFolder = async ({
...
+}): Promise<void> => {
+  const response = await request({
+    method: 'PUT',
+    path: join('remote.php', 'dav', endPath, pathToFile),
+    body: content,
+    user: user,
+    formatJson: false
+  })
+
+  checkResponseStatus(response, `Failed while uploading ${pathToFile} file in personal space`)
```
Comment (saw-jan) [inline tests/e2e/support/api/davSpaces/spaces.ts:74]:
```suggestion
  checkResponseStatus(response, `Failed while uploading file '${pathToFile}' in personal space`)
```

File: tests/e2e/cucumber/steps/api.ts [hunk 2]
```diff
@@ -126,3 +128,35 @@ Given(
...
+)
+
+Given(
+  '{string} uploads following local file in personal space using API',
+  async function (this: World, stepUser: string, stepTable: DataTable): Promise<void> {
+    const user = this.usersEnvironment.getUser({ key: stepUser })
+    const lastIndex = config.assets.lastIndexOf('/')
+    const requiredBasePathToLocalfile = config.assets.slice(0, lastIndex + 1)
+    for (const info of stepTable.hashes()) {
+      const pathToLocalFile = path.join(requiredBasePathToLocalfile, info.localFile)
```
Comment (saw-jan) [inline tests/e2e/cucumber/steps/api.ts:153]: Can this be useful?
https://github.com/owncloud/web/blob/5626822b6004991fa1ad883723bf430a81512a9b/tests/e2e/support/environment/file.ts#L6-L16

Comment (SagarGi) [reply tests/e2e/cucumber/steps/api.ts:153]: i will look into it  thanks !!

File: tests/e2e/cucumber/features/smoke/search.feature [hunk 3]
```diff
@@ -12,14 +12,14 @@ Feature: Search
       | id    |
       | Alice |
       | Brian |
+    And "Brian" creates the following folder in personal space using API
+      | name                 |
+      | new_share_from_brian |
+    And "Brian" uploads following local file in personal space using API
```
Comment (saw-jan) [inline tests/e2e/cucumber/features/smoke/search.feature:18]:
```suggestion
    And "Brian" uploads the following local file in personal space using API
```

**Review by saw-jan — COMMENTED** [review]
LGTM :+1:

File: tests/e2e/cucumber/steps/api.ts [hunk 4]
```diff
@@ -139,3 +140,33 @@ Given(
...
+        user,
+        pathToFile: info.pathToFile,
+        content: info.content
+      })
+    }
+  }
+)
+
+Given(
+  '{string} uploads the following local file(s) in personal space using API',
```
Comment (individual-it) [inline tests/e2e/cucumber/steps/api.ts:159]:
```suggestion
  '{string} uploads the following local file(s) into personal space using API',
```

File: tests/e2e/cucumber/steps/api.ts [hunk 5]
```diff
@@ -139,3 +140,33 @@ Given(
     }
   }
 )
+
+Given(
+  '{string} creates the following file(s) in personal space using API',
+  async function (this: World, stepUser: string, stepTable: DataTable): Promise<void> {
+    const user = this.usersEnvironment.getUser({ key: stepUser })
+    for (const info of stepTable.hashes()) {
+      await api.dav.uploadFileInPersonalSpace({
```
Comment (individual-it) [inline tests/e2e/cucumber/steps/api.ts:149]: as this is a Given step and an API step I think we should at least check the reponse code of the upload. If the upload does not work for some reason the test will fail in a later step and that will be very confusing to debug

File: tests/e2e/cucumber/steps/api.ts [hunk 6]
```diff
@@ -139,3 +140,33 @@ Given(
...
+)
+
+Given(
+  '{string} uploads the following local file(s) in personal space using API',
+  async function (this: World, stepUser: string, stepTable: DataTable): Promise<void> {
+    const user = this.usersEnvironment.getUser({ key: stepUser })
+    for (const info of stepTable.hashes()) {
+      const fileInfo = this.filesEnvironment.getFile({ name: info.localFile.split('/').pop() })
+      const content = fs.readFileSync(fileInfo.path)
+      await api.dav.uploadFileInPersonalSpace({
```
Comment (individual-it) [inline tests/e2e/cucumber/steps/api.ts:165]: same here, please check the return code of the upload operation

File: tests/e2e/cucumber/features/smoke/copyMove.feature [hunk 7]
```diff
@@ -7,17 +7,21 @@ Feature: Copy
     Given "Admin" creates following users
       | id    |
       | Alice |
+    And "Alice" creates the following folder in personal space using API
```
Comment (individual-it) [inline tests/e2e/cucumber/features/smoke/copyMove.feature:10]:
```suggestion
    And "Alice" creates the following folders in personal space using API
```

**Review by individual-it — CHANGES_REQUESTED** [review]

File: tests/e2e/cucumber/steps/api.ts (hunk 5 above)
Comment (SagarGi) [reply tests/e2e/cucumber/steps/api.ts:149]: we have a function checkResponseStatus(response, Failed while uploading file '${pathToFile}' in personal space) that is checked while creating the file which is inside createFile function

**Review by individual-it — APPROVED** [review]

**Review by ScharfViktor — APPROVED** [review]

**Comment by SagarGi** [comment]
TODO:
- [ ] Back port to master branch

**Pull request merged**